### PR TITLE
Add neuralnetwork [Concept Entry]  under TensorFlow

### LINKED ISSUE #4595: [Concept Entry] TensorFlow neuralnetwork
### Reason/inspiration (optional)

We would like a new entry on the `NeuralNetwork` concept under TensorFlow. The entry should go in a new file under `docs/content/tensorflow/concepts/nn/nn.md`.

The entry should include:

- An introductory paragraph that describes the concept
- Explain the nn module in tensorflow

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by vivitt**
### Description

This change adds a new nn.md file under ./content/tensorflow/concepts/nn/nn.md with a [Concept Entry] for neural networks in TensorFlow. 

The entry includes an overview of the neural network concept, a brief description of the layer structure in TensorFlow neural networks, and an enumeration of the steps to build a neural network with TensorFlow.

### Issue Solved

Closes #4595 

### Type of Change

- Adding a New Concept Entry for Existing Topic

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Review by avdhoottt — APPROVED** [review]
Ready for the next review!

**Review by mamtawardhani — APPROVED** [review]
Hey @vivitt thank you for contributing to Codecademy Docs 😄 

The entry is ready to be merged! 🚀
